src/pages/login/login.tsx

```diff
@@ -1,14 +1,66 @@
-import LoginForm from '../../components/blocks/login-form/login-form.tsx';
 import Layout from '../../components/layout/layout.tsx';
+import {FormEvent, useRef} from 'react';
+import {useAppDispatch} from '../../hooks';
+import {loginAction} from '../../store/api-actions.ts';
 
 function Login(): JSX.Element {
+  const loginRef = useRef<HTMLInputElement | null>(null);
+  const passwordRef = useRef<HTMLInputElement | null>(null);
+
+  const dispatch = useAppDispatch();
+  const validPassword = /^(?=.*\d) ((?=.*[a-z]) || (?=.*[A-Z]))$/;
```

Comment:
Лучше немного оптимизировать
```suggestion
  const validPassword = /^(?=.*[a-zA-Z])(?=.*\d).+$/;
;
```